localstack/config.py

```diff
@@ -552,6 +552,11 @@ def in_docker():
 # limit in which to kinesis-mock will start throwing exceptions
 KINESIS_SHARD_LIMIT = os.environ.get("KINESIS_SHARD_LIMIT", "").strip() or "100"
 
+# limit in which to kinesis-mock will start throwing exceptions
+KINESIS_ON_DEMAND_STREAM_COUNT_LIMIT = os.environ.get(
+    "KINESIS_ON_DEMAND_STREAM_COUNT_LIMIT", ""
```

Comment:
for the CLI to pass through the env var correctly to the container, it needs to be whitelisted in the `CONFIG_ENV_VARS` further down in `config.py`

Reply:
Fixed